class-google-spreadsheet.php

```diff
@@ -5,80 +5,132 @@
 
-class Cgs_Google_Spreadsheet {
-	private $token;
-	private $spreadsheet;
-	private $worksheet;
-	const clientId     = '448551536053-e36uicg9npg51m0e89kb51i37b6741fq.apps.googleusercontent.com';
-	const clientSecret = 'kIAz_RGoD6hSGZyYKmIaSZGh';
-	const redirect     = 'urn:ietf:wg:oauth:2.0:oob';
+	class Cgs_Google_Spreadsheet {
 
-	/**
-	* Function Name: google_connect_url
-	* Function Description: Generate google connect url.
-	*/
-	public static function google_connect_url () {
-		$cf7_google_url  = '';
-		$cf7_google_url .= 'https://accounts.google.com/o/oauth2/auth?response_type=code&access_type=offline&';
-		$cf7_google_url .= 'client_id='.Cgs_Google_Spreadsheet::clientId;
-		$cf7_google_url .= '&redirect_uri='.Cgs_Google_Spreadsheet::redirect;
-		$cf7_google_url .= '&state&scope=https://spreadsheets.google.com/feeds';
-		return $cf7_google_url;
-	}
+		/**
+		 * Class instance.
+		 * @since 1.1.0
+		 * @access private
+		 * @var $instance Class instance.
+		 */
+		private static $instance;
 
-	/**
-	* Function Name: google_pre_authentication
-	* Function Description: Pre-authenticate entered access code.
-	* @param object $access_code
-	*/
-	public static function google_pre_authentication( $access_code ) {
-		$client = new Google_Client();
-		$client->setClientId( Cgs_Google_Spreadsheet::clientId );
-		$client->setClientSecret( Cgs_Google_Spreadsheet::clientSecret );
-		$client->setRedirectUri( Cgs_Google_Spreadsheet::redirect );
-		$client->setScopes( array( 'https://spreadsheets.google.com/feeds' ) );
-		try{
-		$results = $client->authenticate( $access_code );
+		/**
+		 * @since 1.1.0
+		 * Initiator
+		 */
+		public static function get_instance() {
+			if ( ! isset( self::$instance ) ) {
+				self::$instance = new self();
+			}
+			return self::$instance;
 		}
-		catch(Exception $e) {
-			global $error_message_code;
-			$error_message_code = '<span style="color:red;">Error: Entered code is incorrect, please enter a valid code.</span></br> </br>';
+	
+		/**
+		 *   @since 1.1.0
+		 *  Constructor
+		 */
+		public function __construct() {
+			$token;
+			$spreadsheet;
+			$worksheet;
+			
 		}
-		$token_data = json_decode( $client->getAccessToken(), true );
-		Cgs_Google_Spreadsheet::update_token( $token_data );
-	}
-	/**
-	* Function Name: update_token
-	* Function Description: Updating access token .
-	* @param object $token_data
-	*/
-	public static function update_token( $token_data ) {
-		$token_data['expire'] = time() + intval( $token_data['expires_in'] );
-		try{
-			$tokenJson = json_encode( $token_data );
-			update_option( 'cf7_to_spreadsheet_google_token', $tokenJson );
-		} catch ( Exception $e ) {
-			echo "error";
+
+		const redirect     = 'urn:ietf:wg:oauth:2.0:oob';
+
+		/**
+		* Function Name: get_api_key
+		* @since 1.1.0
+		* @return api key array $data 
+		* Function Description: Get the api key from database.
+		*/
+		public static function get_api_key(){
+			$clientid = get_option('clientid');
```

Comment:
Add a prefix `cf7_to_spreadsheet_` for this option to avoid possible name collision with any other plugins.
be sure to udpate this option name everywhere in codebase.